src/js/components/Image/doc.js

```diff
@@ -26,6 +26,9 @@ export const doc = Image => {
       PropTypes.string,
       PropTypes.bool,
     ]).description('Transparency of the image.'),
+    fillProp: PropTypes.bool.description(
```

Comment:
I think the prop should still be called `fill` for the users (consistent across grommet) and it should use the `fillProp` internally as an alias.